src/Components/Endpoints/src/Builder/RazorComponentsEndpointRouteBuilderExtensions.cs

```diff
@@ -26,10 +29,42 @@ public static RazorComponentEndpointConventionBuilder MapRazorComponents<TRootCo
         ArgumentNullException.ThrowIfNull(endpoints);
 
         EnsureRazorComponentServices(endpoints);
+        AddBlazorWebJsEndpoint(endpoints);
 
         return GetOrCreateDataSource<TRootComponent>(endpoints).DefaultBuilder;
     }
 
+    private static void AddBlazorWebJsEndpoint(IEndpointRouteBuilder endpoints)
+    {
+        var options = new StaticFileOptions
+        {
+            FileProvider = new ManifestEmbeddedFileProvider(typeof(RazorComponentsEndpointRouteBuilderExtensions).Assembly),
+            OnPrepareResponse = CacheHeaderSettings.SetCacheHeaders
+        };
+
+        var app = endpoints.CreateApplicationBuilder();
+        app.Use(next => context =>
+        {
+            // Set endpoint to null so the static files middleware will handle the request.
+            context.SetEndpoint(null);
+
+            return next(context);
+        });
+        app.UseStaticFiles(options);
+
+        var blazorEndpoint = endpoints.Map("/_framework/blazor.web.js", app.Build())
```

Comment:
How do we feel about this being just `blazor.js` and leaving the other more specialized flavors with a suffix.